ocs_ci/helpers/mcg_stress_helper.py

```diff
@@ -0,0 +1,618 @@
+                    f"\nScaled noobaa db to new size {new_size}"
+                    f"\n"
+                )
+            logger.info(
+                f"\n"
+                f"\n[BACKGROUND CHECK]"
+                f"\nCurrent noobaa db usage is at {used_percent}%... Rechecking in 10 minutes..."
+                f"\n"
+            )
+
+            if event.is_set():
+                logger.info("[BACKGROUND CHECK] Stopping the Noobaa db size check")
+                break
+            time.sleep(600)
+
+    def check_prometheus_alerts():
+
+        while True:
+            prometheus_alert_list = list()
+            prometheus_api.prometheus_log(prometheus_alert_list)
+            alert_tab = list()
+            alert_printed = list()
+            alert_tab.append(["Alert Name", "Description", "State"])
+            for alert in prometheus_alert_list:
+                if alert["labels"]["alertname"] in alert_printed:
+                    continue
+                alert_tab.append(
+                    [
+                        alert["labels"]["alertname"].strip(),
+                        "\n".join(
+                            textwrap.wrap(alert["annotations"]["description"], width=50)
+                        ),
+                        alert["state"],
+                    ]
+                )
+                alert_printed.append(alert["labels"]["alertname"])
+            logger.info(
+                f"\n"
+                f"\n[BACKGROUND CHECK]"
+                f"\nThese are the alerts so far in Prometheus: "
+                f"\n{tabulate(alert_tab[1:], headers=alert_tab[0], tablefmt='grid')}"
+                f"\n"
+            )
+
+            if event.is_set():
+                logger.info("[BACKGROUND CHECK] Stopping Prometheus alert logging")
+                break
+            time.sleep(300)
+
+    def check_noobaa_pod_resource_utilization():
+
+        while True:
+            logger.info(
+                f"\n"
+                f"\n[BACKGROUND CHECK]"
+                f"\nCurrent noobaa pod resource utilization: "
+                f"\n{pod_resource_utilization_raw_output_from_adm_top(selector=constants.NOOBAA_APP_LABEL)}"
+                f"\n"
+            )
+
+            if event.is_set():
+                logger.info(
+                    "[BACKGROUND CHECK] Stopping noobaa pod resource utilization checks"
+                )
+                break
+            time.sleep(600)
+
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=5)
+    futures_obj = list()
+    futures_obj.append(executor.submit(check_noobaa_health))
+    futures_obj.append(executor.submit(check_ceph_health))
+    futures_obj.append(executor.submit(check_noobaa_db_size))
+    futures_obj.append(executor.submit(check_prometheus_alerts))
+    futures_obj.append(executor.submit(check_noobaa_pod_resource_utilization))
+    for future in as_completed(futures_obj):
+        future.result()
+    executor.shutdown()
```

Comment:
you can consider calling using with statement to ensure that the executor is  automatically shut downed. that way you don't need to manually handle it here

Reply:
@PrasadDesala will address this in the next pr